File: examples/2. puzzle/puzzle.pl
:- use_module(library('bounds')).

solusi(Papan) :-
	Papan = [NW,N,NE,W,E,SW,S,SE],
	Papan in 0..12,
	sum(Papan, #=, 15),
	NW + N + NE #= 6,
	NE + E + SE #= 6,
	NW + W + SW #= 6,
	SW + S + SE #= 6,

	label(Papan).

File: examples/2. puzzle/puzzle.py


#Kemungkinan dalam papan puzzle (8 kotak) dimana
#dalam satu baris/garis jumlah angkanya harus 6 dan
#total semua angka dari papan harus 15


from pyswip.prolog import Prolog     #import library pyswip


def main():
    prolog = Prolog()
    prolog.consult("puzzle.pl")            #membaca file prolog (puzzle.pl)
    for soln in prolog.query("solusi(B)."):
        #B = eval(soln["B"])
        B = soln["B"]
        # [NW,N,NE,W,E,SW,S,SE]
        print ("%d %d %d" % tuple(B[:3]))
        print ("%d   %d"   % tuple(B[3:5]))
        print ("%d %d %d" % tuple(B[5:]))
        
        cont = input("Tekan 'n' untuk selesai: ")
        if cont.lower() == "n": break

        
if __name__ == "__main__":
    main()
